test/tpmtest/tpmtest.c

```diff
@@ -458,6 +460,95 @@ void GetTpmVersion()
     }
 }
 
+// Get the TPM 2.0 commands supported by the TPM.
+TSS2_RC GetCommands( TSS2_SYS_CONTEXT *resMgrSysContext, TPML_CCA **supportedCommands )
+{
+    UINT32 numCommands;
+    TPMI_YES_NO moreData;
+    TPMS_CAPABILITY_DATA capabilityData;
+    TPMA_CC *commandPtr;
+    TSS2_RC rval = TSS2_RC_SUCCESS;
+    UINT32 i;
+
+    // First get the number of commands
+    rval = Tss2_Sys_GetCapability( resMgrSysContext, 0,
+            TPM_CAP_TPM_PROPERTIES, TPM_PT_TOTAL_COMMANDS,
+            1, 0, &capabilityData, 0 );
+    if( rval == TPM_RC_SUCCESS &&
+            capabilityData.capability == TPM_CAP_TPM_PROPERTIES &&
+            capabilityData.data.tpmProperties.count == 1 &&
+            capabilityData.data.tpmProperties.tpmProperty[0].property == TPM_PT_TOTAL_COMMANDS )
+    {
+        numCommands = capabilityData.data.tpmProperties.tpmProperty[0].value;
+    }
+    else
+    {
+        goto returnFromGetCommands;
+    }
+
+    // Allocate memory for them
+    *supportedCommands = (TPML_CCA *)malloc( numCommands * sizeof( TPMA_CC ) + sizeof( UINT32 ) );
+    if( !*supportedCommands )
+    {
+        rval = TSS2_BASE_RC_INSUFFICIENT_BUFFER + TSS2_RESMGR_ERROR_LEVEL;
+        goto returnFromGetCommands;
+    }
+
+    for( commandPtr = &( ( *supportedCommands )->commandAttributes[0] ), ( *supportedCommands )->count = 0;
+            ( *supportedCommands )->count < numCommands;
+            ( *supportedCommands )->count += capabilityData.data.command.count,
+                    commandPtr += capabilityData.data.command.count )
+    {
+        // Now get the command structures for all of them.
+        rval = Tss2_Sys_GetCapability( resMgrSysContext, 0,
+                TPM_CAP_COMMANDS, TPM_CC_FIRST,
+                numCommands, &moreData, &capabilityData, 0 );
+
+        if( rval == TPM_RC_SUCCESS &&
+                capabilityData.capability == TPM_CAP_COMMANDS  &&
+                capabilityData.data.command.count >= 1 )
+        {
+            for( i = 0; i < capabilityData.data.command.count; i ++ )
+            {
+                commandPtr[ ( *supportedCommands )->count + i ].val =
+                        capabilityData.data.command.commandAttributes[i].val;
+            }
+        }
+        else
+        {
+            break;
+        }
+    }
+
+returnFromGetCommands:
+    return rval;
+}
+
+//
+// Searches in a list for command attributes bit field for a command code.
+// Assumes that the supportedCommands structure has been populated by
+// calling GetCommands beforehand.
+//
+// Returns:
+//  1, if found.
+//  0, if not found.  This means that the TPM doesn't support this command.
+//
+UINT8 IsCommandImpl( TPM_CC commandCode, TPML_CCA *supportedCommands )
```

Comment:
The semantics of this call (using the "Is" prefix) makes me think it should return a boolean value. Doing so will make code that calls the function much easier to read:

```
if (IsCommandImplemented (TPM_CC_ChangeEPS, supportedCommands))
{
    /* do stuff */
}
```